miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultBoxService.java

```diff
@@ -264,13 +277,44 @@ public long save(Box box) throws IOException {
     }
   }
 
+  /**
+   * Checks submitted data for validity, throwing a ValidationException containing all of the errors if invalid
+   * 
+   * @param box submitted Box to validate
+   * @param beforeChange the already-persisted Box before changes
+   * @throws IOException
+   */
+  private void validateChange(Box box, Box beforeChange) throws IOException {
+    List<ValidationError> errors = new ArrayList<>();
+
+    if (box.getStorageLocation() != null) {
+      if (box.getStorageLocation().getLocationUnit().getBoxStorageAmount() == BoxStorageAmount.NONE) {
+        errors.add(new ValidationError("storageLocation", "Invalid box location"));
+      } else if (box.getStorageLocation().getLocationUnit().getBoxStorageAmount() == BoxStorageAmount.SINGLE) {
+        if (beforeChange == null || beforeChange.getStorageLocation() == null
+            || beforeChange.getStorageLocation().getId() != box.getStorageLocation().getId()) {
+          box.getStorageLocation().getBoxes().forEach(b -> {
+            if (b.getId() != box.getId()) {
+              errors.add(new ValidationError("storageLocation", "Location already occupied"));
+            }
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)